components/apimgt/org.wso2.carbon.apimgt.persistence.mongodb/src/main/java/org/wso2/carbon/apimgt/persistence/mongodb/MongoDBPersistenceImpl.java

```diff
@@ -0,0 +1,813 @@
+        orMatchList.add(matchPublished);
+        orMatchList.add(matchPrototyped);
+        orDoc.put("$or", orMatchList);
+
+        matchExists.put("$exists", false);
+        matchRevisions.put("revision", matchExists);
+
+        andMatchList.add(orDoc);
+        andMatchList.add(matchRevisions);
+
+        andDoc.put("$and", andMatchList);
+        matchDoc.put("$match", andDoc);
+
+        List<Document> list = new ArrayList<>();
+        list.add(search);
+        list.add(matchDoc);
+        list.add(skipDoc);
+        list.add(limitDoc);
+        return list;
+    }
+
+    @Override
+    public DevPortalAPISearchResult searchAPIsForDevPortal(Organization org, String searchQuery, int start,
+                                                           int offset, UserContext ctx) throws APIPersistenceException {
+        int skip = start;
+        int limit = offset;
+        //published prototyped only
+        searchQuery = "";
+        MongoCollection<MongoDBDevPortalAPI> collection = getDevPortalCollection(ctx.getOrganization().getName());
+        long totalCount = collection.countDocuments();
+        MongoCursor<MongoDBDevPortalAPI> aggregate = collection.aggregate(getDevportalSearchAggregate(searchQuery, skip, limit))
+                .cursor();
+        DevPortalAPISearchResult devPortalAPISearchResult = new DevPortalAPISearchResult();
+        List<DevPortalAPIInfo> devPortalAPIInfoList = new ArrayList<>();
+        while (aggregate.hasNext()) {
+            MongoDBDevPortalAPI mongoDBAPIDocument = aggregate.next();
+            DevPortalAPI api = MongoAPIMapper.INSTANCE.toDevPortalApi(mongoDBAPIDocument);
+            devPortalAPIInfoList.add(api);
+        }
+        devPortalAPISearchResult.setDevPortalAPIInfoList(devPortalAPIInfoList);
+        devPortalAPISearchResult.setReturnedAPIsCount(devPortalAPIInfoList.size());
+        devPortalAPISearchResult.setTotalAPIsCount((int) totalCount);
+        return devPortalAPISearchResult;
+    }
+
+    @Override
+    public void changeAPILifeCycle(Organization org, String apiId, String status)
+            throws APIPersistenceException {
+        MongoCollection<MongoDBPublisherAPI> collection = getPublisherCollection(org.getName());
+        collection.updateOne(eq("_id", new ObjectId(apiId)), set("status", status));
+    }
+
+    @Override
+    public void saveWSDL(Organization org, String apiId, ResourceFile wsdlResourceFile)
+            throws WSDLPersistenceException {
+
+    }
+
+    @Override
+    public ResourceFile getWSDL(Organization org, String apiId) throws WSDLPersistenceException {
+        return null;
+    }
+
+    @Override
+    public void saveOASDefinition(Organization org, String apiId, String apiDefinition)
+            throws OASPersistenceException {
+        MongoCollection<MongoDBPublisherAPI> collection = getPublisherCollection(org.getName());
+        collection.updateOne(eq("_id", new ObjectId(apiId)), set("swaggerDefinition", apiDefinition));
```

Comment:
shall we rename this to "oasDefinition" instead of "swaggerDefinition"

Reply:
swaggerDefinition is the name of the field in publisher api object, changing this means i should change that as well, is that ok?